db/commitlog/commitlog.cc

```diff
@@ -1779,6 +1780,76 @@ void db::commitlog::segment_manager::flush_segments(uint64_t size_to_remove) {
     totals.bytes_flush_requested += flushing;
 }
 
+void db::commitlog::segment_manager::check_no_data_older_than_allowed() {
+    if (!cfg.commitlog_data_max_lifetime_in_seconds) {
+        return;
+    }
+
+    auto now = gc_clock::now();
+
+    std::unordered_set<cf_id_type> ids;
+    std::optional<replay_position> high;
+
+    for (auto& s : _segments) {
+        auto rp = replay_position(s->_desc.id, db::position_type(s->size_on_disk()));
+        if (rp <= _flush_position) {
+            // already requested.
+            continue;
+        }
+
+        bool any_found = false;
+
+        for (auto [id, low_ts] : s->_cf_min_time) {
+            // Ignore CF:s already released.
+            if (!s->_cf_dirty.count(id)) {
+                continue;
+            }
+
```

Comment:
can a segment maintain the lowest ts, such that we can quickly rule out a segment without having to iterate through CFs (s->_cf_min_time)? so for the common scenario where segments aren't getting too old, we have a fast path.

Reply:
I guess. It adds more work to "add" (writing mutations to commit log), though it should be negligible. Would also need to recompute on every clear, but also fairly light weight.